src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3614,7 +3547,18 @@ internal static Result Create(TypeSymbolWithAnnotations type, int slot = -1)
                 return new Result(null, type, slot);
             }
 
-            private Result(BoundExpression expr, TypeSymbolWithAnnotations type, int slot)
+            public static implicit operator Result(BoundExpression expr)
```

Comment:
I'm not sure how I feel about using conversion operator here. 
Feels like overloading the constructor (`new Result(expr)` and `new Result(type)`) would work fine and the logic would be more readable.

Reply:
I'm not sold on using implicit conversion operators here either, although it did simplify the code and the operators are local to `NullableWalker`. For now, I've added a `// PROTOTYPE` comment to consider replacing with either `Result.Create()` overloads or `ToResult()` extension methods as we discussed.